commons/src/main/java/org/infinispan/commons/io/UnsignedNumeric.java

```diff
@@ -170,12 +170,14 @@ public static int readUnsignedInt(byte[] bytes, int offset) {
     *
     * @param i int to write
     */
-   public static void writeUnsignedInt(byte[] bytes, int offset, int i) {
+   public static int writeUnsignedInt(byte[] bytes, int offset, int i) {
```

Comment:
> I wonder if returning the size used may be better, WDYT?

I've tried that but makes the client usage more complex since you need to keep both written bytes and offset separately. Offset is needed to figure out where to start writing in the byte[]. Returning the new offset makes it simpler IMO.